synapse/handlers/message.py

```diff
@@ -790,7 +794,10 @@ async def create_new_client_event(
             Tuple of created event, context
         """
 
+        overriding_prev_events = False
         if prev_event_ids is not None:
+            overriding_prev_events = True
```

Comment:
This code around `overriding_prev_events` and later in `synapse/events/builder.py` make the DAG look correct and respond correctly from `/messages` :tada:, see https://github.com/matrix-org/complement/pull/68#discussion_r568507353